### ISSUE TITLE
Adding multiple documents should not use MULTI

### ISSUE THREAD

**Issue by dvirsky** [vruizext/redisearch-rb#1]
In https://github.com/vruizext/redisearch-rb/blob/master/lib/redisearch.rb#L89 adding multiple docs uses a MULTI transaction. 

It can cause redis to crash on 4.0.2, and trigger an error on unstable, but FT.ADD is a blocking command and cannot be called in MULTI. You should use a normal pipeline. 

See https://github.com/RedisLabsModules/RediSearch/issues/179

**Comment by id4coding** [comment]
Thanks for opening this issue. A fix to this issue is eagerly awaited for making our RoR server work with RediSearch & Redis.

**Comment by dvirsky** [comment]
@id4coding in the mean time index document by document, it doesn't use multi. use add_doc instead of add_docs. It will be slower but will work

**Comment by id4coding** [comment]
@dvirsky thanks for pointing that out. I'm using add_doc instead of add_docs. But, will add_docs be supported going forward?

**Comment by dvirsky** [comment]
Actually yes, as a side effect of another fix, there will be a second indexing command that works with multi. However, it will not allow for concurrent indexing, so it might affect your app if you are doing a lot of concurrent reads and writes.

**Comment by vruizext** [comment]
@id4coding I'm sorry for the delay in the answer. Will address this soon. 

Btw PRs are welcome :)

**Comment by vruizext** [comment]
To not make it much slower, we can still use `MULTI` to add the hashes to redis, and then use a standard pipeline to index the docs using `FT.ADDHASH`. WYT @dvirsky ?

```ruby
  def add_docs(docs, opts = {})
    multi { docs.each { |doc_id, fields| @redis.hmset(doc_id, *fields) } }
    docs.each { |doc_id, _| call(ft_add_hash(doc_id, opts))}
  end
``` 

```ruby
  def ft_add_hash(doc_id, opts = {}, weight =  nil)
    ['FT.ADDHASH', @idx_name , doc_id, weight || DEFAULT_WEIGHT, *add_options(opts)]
  end
```

**Comment by vruizext** [comment]
Fixed in 0917fa2910956e9bfd0e28016fa881183307316c
